lib/idempotency.rb

```diff
@@ -5,12 +5,22 @@
 require 'base64'
 require_relative 'idempotency/cache'
 require_relative 'idempotency/constants'
+require_relative 'idempotency/instrumentation/datadog_listener'
+require 'dry-monitor'
 
 class Idempotency
   extend Dry::Configurable
 
+  def self.notifier
+    @notifier ||= Dry::Monitor::Notifications.new(:idempotency_gem).tap do |n|
+      Events::ALL_EVENTS.each { |event| n.register_event(event) }
+    end
+  end
```

Comment:
Using class instance variable may not be thread-safe 🤔

Should we wrap mutex around it?